docs/types/value-types.rst

```diff
@@ -494,21 +496,22 @@ regardless of the type of the right (exponent) operand.
     Division on integer literals used to truncate in Solidity prior to version 0.4.0, but it now converts into a rational number, i.e. ``5 / 2`` is not equal to ``2``, but to ``2.5``.
 
 .. note::
-    Solidity has a number literal type for each rational number.
-    Integer literals and rational number literals belong to number literal types.
+    In solidity, rational numbers and integers expressed as literals are of number literal type.
     Moreover, all number literal expressions (i.e. the expressions that
-    contain only number literals and operators) belong to number literal
-    types.  So the number literal expressions ``1 + 2`` and ``2 + 1`` both
-    belong to the same number literal type for the rational number three.
+    contain only number literals and operators) are of number literal
+    type.  The number literal expressions ``1 + 2`` and ``2 + 1`` are both
+    of number literal type, just the same as ``3``. Technically speaking, the internal representation
+    for fractional rational numbers like ``1.5`` is different than that of ``3``, specifically to support
+    arbitrary precision. However, all number literals are treated as effectively the same type.
```

Comment:
It's not different. They're both internally represented using the same multi-precision type. If you took the same number as a fractional vs integer (e,g `3.0` vs `3`) there would be absolutely no difference internally. You can consider them not just convertible but exactly equivalent for all intents and purposes is Solidity.

This is really getting into compiler's implementation details that are not relevant to understanding semantics of literals. I'd remove this part.